Print Error Messages to `stderr`

Originally, the issue was reported [here](https://github.com/yegor256/jaxec/issues/57).

Recently, I got the following stack trace:

```java
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 8.464 s <<< FAILURE! -- in integration.JarIT
[ERROR] integration.JarIT.runsProgramFromJar(Path) -- Time elapsed: 7.840 s <<< ERROR!
java.lang.IllegalArgumentException: Non-zero exit code #1 of 'java'
	at com.yegor256.Jaxec.execUnsafe(Jaxec.java:366)
	at com.yegor256.Jaxec.exec(Jaxec.java:329)
	at integration.JarIT.lambda$runsProgramFromJar$0(JarIT.java:75)
	at com.yegor256.farea.Farea.together(Farea.java:179)
	at integration.JarIT.runsProgramFromJar(JarIT.java:35)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   JarIT.runsProgramFromJar:35->lambda$runsProgramFromJar$0:75 » IllegalArgument Non-zero exit code #1 of 'java'
```

As you can see, it's impossible to understand anything from it.

After [checking the logs](https://github.com/yegor256/jaxec/issues/57#issuecomment-2901395124), we have found:

```
DEBUG c.j.l.VerboseProcess - >> Couldn't find object 'Φ.org.eolang.io' because there's no class or package 'EOorg.EOeolang.EOio'
DEBUG c.j.l.VerboseProcess - >> EOorg.EOeolang.EOio
```

It means that `eo` logged the error message, but logged it to stdout instead of stderr which made it [hard to detect](https://github.com/yegor256/jaxec/issues/57#issuecomment-2903582601).

**Expected:**
I believe we should print all the error messages to stderr instead.

--- a/eo-integration-tests/src/test/java/integration/JarIT.java
+++ b/eo-integration-tests/src/test/java/integration/JarIT.java
@@ -29,77 +29,113 @@ import org.junit.jupiter.api.extension.ExtendWith;
 @SuppressWarnings("JTCOP.RuleAllTestsHaveProductionClass")
 @ExtendWith(MktmpResolver.class)
 final class JarIT {
+
     @Test
     @ExtendWith(WeAreOnline.class)
     @ExtendWith(MayBeSlow.class)
     void runsProgramFromJar(final @Mktmp Path temp) throws IOException {
         new Farea(temp).together(
             f -> {
-                f.properties()
-                    .set("project.build.sourceEncoding", StandardCharsets.UTF_8.name())
-                    .set("project.reporting.outputEncoding", StandardCharsets.UTF_8.name());
-                f.files()
-                    .file("src/main/eo/simple.eo")
-                    .write(
-                        String.join(
-                            "\n",
-                            new String[]{
-                                "# No comments.",
-                                "[] > simple",
-                                "  QQ.io.stdout > @",
-                                "    \"Hello, world!\"",
-                            }
-                        ).getBytes(StandardCharsets.UTF_8)
-                    );
-                f.dependencies()
-                    .append(
-                        "org.eolang",
-                        "eo-runtime",
-                        System.getProperty(
-                            "eo.version",
-                            Manifests.read("EO-Version")
-                        )
-                    );
-                new EoMavenPlugin(f)
-                    .appended()
-                    .execution("compile")
-                    .goals("register", "compile", "transpile")
-                    .configuration()
-                    .set("ignoreRuntime", Boolean.TRUE.toString())
-                    .set("failOnWarning", Boolean.FALSE.toString())
-                    .set("skipLinting", Boolean.TRUE.toString());
-                f.exec("clean", "compile", "jar:jar");
-                MatcherAssert.assertThat(
-                    "Project must be successfully built and packaged into jar",
-                    f.log(),
-                    RequisiteMatcher.SUCCESS
-                );
-                final String ver = System.getProperty("eo.version", Manifests.read("EO-Version"));
-                final String jar = String.format("eo-runtime-%s.jar", ver);
-                final String runtime = Paths.get(System.getProperty("user.home")).resolve(".m2")
-                    .resolve("repository")
-                    .resolve("org/eolang/eo-runtime")
-                    .resolve(ver)
-                    .resolve(jar)
-                    .toString();
-                final String classpath = String.join(
-                    File.pathSeparator,
-                    "test-0.0.0.jar",
-                    runtime
+                final String classpath = JarIT.compile(
+                    f,
+                    "# No comments.",
+                    "[] > simple",
+                    "  QQ.io.stdout > @",
+                    "    \"Hello, world!\""
                 );
                 MatcherAssert.assertThat(
-                    "Simple program must be successfully executed from jar",
+                    "simple program must be successfully executed from jar",
                     new Jaxec(
                         "java", "-cp", classpath,
                         "-Dfile.encoding=UTF-8", "-Xss64M", "-Xms64M",
                         "org.eolang.Main", "simple"
                     ).withHome(temp.resolve("target")).exec().stdout(),
+                    Matchers.containsString("Hello, world!")
+                );
+            }
+        );
+    }
+
+    @Test
+    @ExtendWith(WeAreOnline.class)
+    @ExtendWith(MayBeSlow.class)
+    void printsErrorToStderr(final @Mktmp Path temp) throws IOException {
+        new Farea(temp).together(
+            f -> {
+                final String classpath = JarIT.compile(
+                    f,
+                    "# No comments.",
+                    "[] > simple",
+                    "  unknown.io.stdout > @",
+                    "    \"Hello, world!\""
+                );
+                MatcherAssert.assertThat(
+                    "the program must throw an error and print it to stderr",
+                    new Jaxec(
+                        "java", "-cp", classpath,
+                        "-Dfile.encoding=UTF-8", "-Xss64M", "-Xms64M",
+                        "org.eolang.Main", "simple"
+                    ).withHome(temp.resolve("target")).withCheck(false).execUnsafe().stderr(),
                     Matchers.allOf(
-                        Matchers.containsString("Hello, world!"),
-                        Matchers.containsString("[0x01] = true")
+                        Matchers.containsString("Couldn't find object 'Φ.org.eolang.unknown'"),
+                        Matchers.containsString(
+                            "because there's no class or package 'EOorg.EOeolang.EOunknown"
+                        )
                     )
                 );
             }
+        );
+    }
+
+    /**
+     * Compile EO program to XMIR and package it into a JAR.
+     * @param farea Farea to use for compilation
+     * @param program The EO program to compile
+     * @return Classpath for the compiled program, with eo-runtime jar
+     * @throws IOException If fails to compile
+     */
+    private static String compile(final Farea farea, final String... program) throws IOException {
+        farea.properties()
+            .set("project.build.sourceEncoding", StandardCharsets.UTF_8.name())
+            .set("project.reporting.outputEncoding", StandardCharsets.UTF_8.name());
+        farea.files()
+            .file("src/main/eo/simple.eo")
+            .write(String.join("\n", program).getBytes(StandardCharsets.UTF_8));
+        farea.dependencies()
+            .append(
+                "org.eolang",
+                "eo-runtime",
+                System.getProperty(
+                    "eo.version",
+                    Manifests.read("EO-Version")
+                )
+            );
+        new EoMavenPlugin(farea)
+            .appended()
+            .execution("compile")
+            .goals("register", "compile", "transpile")
+            .configuration()
+            .set("ignoreRuntime", Boolean.TRUE.toString())
+            .set("failOnWarning", Boolean.FALSE.toString())
+            .set("skipLinting", Boolean.TRUE.toString());
+        farea.exec("clean", "compile", "jar:jar");
+        MatcherAssert.assertThat(
+            "Project must be successfully built and packaged into jar",
+            farea.log(),
+            RequisiteMatcher.SUCCESS
+        );
+        final String ver = System.getProperty("eo.version", Manifests.read("EO-Version"));
+        final String jar = String.format("eo-runtime-%s.jar", ver);
+        final String runtime = Paths.get(System.getProperty("user.home")).resolve(".m2")
+            .resolve("repository")
+            .resolve("org/eolang/eo-runtime")
+            .resolve(ver)
+            .resolve(jar)
+            .toString();
+        return String.join(
+            File.pathSeparator,
+            "test-0.0.0.jar",
+            runtime
         );
     }
 }

--- a/eo-integration-tests/src/test/resources/logback.xml
+++ b/eo-integration-tests/src/test/resources/logback.xml
@@ -5,6 +5,7 @@
 -->
 <configuration>
   <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+    <target>System.out</target>
     <encoder>
       <pattern>%highlight(%-5level) %logger{15} - %msg%n</pattern>
     </encoder>

--- a/eo-runtime/src/main/java/org/eolang/Dataized.java
+++ b/eo-runtime/src/main/java/org/eolang/Dataized.java
@@ -29,7 +29,7 @@ import java.util.logging.Logger;
 @SuppressWarnings("java:S5164")
 public final class Dataized {
     /**
-     * The object to datarize.
+     * The object to dataize.
      */
     private final Phi phi;
 

--- a/eo-runtime/src/main/java/org/eolang/Main.java
+++ b/eo-runtime/src/main/java/org/eolang/Main.java
@@ -56,8 +56,7 @@ public final class Main {
      */
     public static void main(final String... args) throws Exception {
         Main.setup();
-        final List<String> opts = new ArrayList<>(args.length);
-        opts.addAll(Arrays.asList(args));
+        final List<String> opts = new ArrayList<>(Arrays.asList(args));
         while (!opts.isEmpty()) {
             final String opt = opts.get(0);
             if (Main.parse(opt)) {
@@ -151,9 +150,8 @@ public final class Main {
     /**
      * Run this opts.
      * @param opts The opts left
-     * @throws Exception If fails
      */
-    private static void run(final List<String> opts) throws Exception {
+    private static void run(final List<String> opts) {
         final String obj = opts.get(0);
         if (obj.isEmpty()) {
             throw new IllegalArgumentException(
@@ -187,13 +185,14 @@ public final class Main {
      * @throws IOException If fails
      */
     private static String version() throws IOException {
-        try (BufferedReader input =
-            new BufferedReader(
-                new InputStreamReader(
-                    Objects.requireNonNull(Main.class.getResourceAsStream("version.txt")),
-                    StandardCharsets.UTF_8
+        try (
+            BufferedReader input =
+                new BufferedReader(
+                    new InputStreamReader(
+                        Objects.requireNonNull(Main.class.getResourceAsStream("version.txt")),
+                        StandardCharsets.UTF_8
+                    )
                 )
-            )
         ) {
             return input.lines().findFirst().orElse("N/A");
         }

--- a/eo-runtime/src/test/java/org/eolang/MainTest.java
+++ b/eo-runtime/src/test/java/org/eolang/MainTest.java
@@ -6,6 +6,7 @@ package org.eolang;
 
 import com.yegor256.Jaxec;
 import com.yegor256.Jhome;
+import com.yegor256.Result;
 import java.io.BufferedReader;
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
@@ -41,7 +42,7 @@ final class MainTest {
     void printsVersion() {
         MatcherAssert.assertThat(
             "Prints its own version properly",
-            MainTest.exec("--version"),
+            MainTest.stderr("--version"),
             Matchers.allOf(
                 Matchers.containsString("."),
                 Matchers.not(Matchers.containsString(" "))
@@ -53,7 +54,7 @@ final class MainTest {
     void printsHelp() {
         MatcherAssert.assertThat(
             "Prints help summary properly",
-            MainTest.exec("--help"),
+            MainTest.stderr("--help"),
             Matchers.containsString("Usage: ")
         );
     }
@@ -62,7 +63,7 @@ final class MainTest {
     void deliversCleanOutput() {
         MatcherAssert.assertThat(
             "Incorrect output when dataizing \"true\" object",
-            MainTest.exec("org.eolang.true"),
+            MainTest.stderr("org.eolang.true"),
             Matchers.stringContainsInOrder(
                 String.format("%n---%n"),
                 "true",
@@ -75,7 +76,7 @@ final class MainTest {
     void executesJvmFullRun() {
         MatcherAssert.assertThat(
             "Incorrect verbose output when dataizing \"false\" object",
-            MainTest.exec("--verbose", "org.eolang.false"),
+            MainTest.stderr("--verbose", "org.eolang.false"),
             Matchers.allOf(
                 Matchers.containsString("EOLANG"),
                 Matchers.containsString("false")
@@ -87,7 +88,7 @@ final class MainTest {
     void executesJvmFullRunWithDashedObject() {
         MatcherAssert.assertThat(
             "Fails with the proper error message",
-            MainTest.exec("--verbose", "as-bytes"),
+            MainTest.stderr("--verbose", "as-bytes"),
             Matchers.containsString("Couldn't find object 'Φ.as-bytes'")
         );
     }
@@ -96,7 +97,7 @@ final class MainTest {
     void executesJvmFullRinWithAttributeCall() {
         MatcherAssert.assertThat(
             "Fails with the proper error message",
-            MainTest.exec("--verbose", "string$as-bytes"),
+            MainTest.stderr("--verbose", "string$as-bytes"),
             Matchers.containsString("Couldn't find object 'Φ.string$as-bytes'")
         );
     }
@@ -105,7 +106,7 @@ final class MainTest {
     void executesJvmFullRunWithError() {
         MatcherAssert.assertThat(
             "Fails with the proper error message",
-            MainTest.exec("--verbose", "org.eolang.io.stdout"),
+            MainTest.stderr("--verbose", "org.eolang.io.stdout"),
             Matchers.containsString(
                 "Error in \"Φ.org.eolang.io.stdout.φ.Δ\" "
             )
@@ -116,7 +117,7 @@ final class MainTest {
     void executesWithObjectNotFoundException() {
         MatcherAssert.assertThat(
             "Fails with the proper error message",
-            MainTest.exec("unavailable-name"),
+            MainTest.stderr("unavailable-name"),
             Matchers.containsString("Couldn't find object 'Φ.unavailable-name'")
         );
     }
@@ -176,8 +177,12 @@ final class MainTest {
         );
     }
 
-    private static String exec(final String... cmds) {
-        final String stdout = new Jaxec(
+    private static String stderr(final String... cmds) {
+        return MainTest.exec(cmds).stderr();
+    }
+
+    private static Result exec(final String... cmds) {
+        return new Jaxec(
             new Jhome().java().toString(),
             "-Dfile.encoding=UTF-8",
             "-Dsun.stdout.encoding=UTF-8",
@@ -185,14 +190,7 @@ final class MainTest {
             "-cp",
             System.getProperty("java.class.path"),
             Main.class.getCanonicalName()
-        ).with(cmds).withCheck(false).exec().stdout();
-        return stdout.replaceFirst(
-            String.format(
-                "Picked up .*%s",
-                System.lineSeparator()
-            ),
-            ""
-        );
+        ).with(cmds).withCheck(false).exec();
     }
 
 }

--- a/pom.xml
+++ b/pom.xml
@@ -202,7 +202,7 @@
       <dependency>
         <groupId>com.yegor256</groupId>
         <artifactId>jaxec</artifactId>
-        <version>0.4.0</version>
+        <version>0.4.1</version>
       </dependency>
       <dependency>
         <groupId>com.yegor256</groupId>

